lib/rules/no-top-level-side-effects.ts

```diff
@@ -290,6 +291,29 @@ export const noTopLevelSideEffects: Rule.RuleModule = {
         }
       },
 
+      ObjectExpression: (node) => {
+        if (isTopLevel(node)) {
+          node.properties.forEach(property => {
+            if (property.type === 'SpreadElement') {
+              checkExpression(context, options, node, property.argument);
+              return;
+            }
+
+            switch (property.value.type) {
+              case 'Identifier':
+              case 'ObjectPattern':
+              case 'ArrayPattern':
+              case 'RestElement':
+              case 'AssignmentPattern':
+              case 'MemberExpression':
+                return;
```

Comment:
It looks like this is where coverage is missing. It seems the first and last aren't covered by tests.

Just having a quick look, the other's seem to be here because they don't have `property.value` on line 311, correct?

Reply:
Superseded by #1095